wand2.c

```diff
@@ -847,7 +847,7 @@ fsize(FILE *fp)
 {
 	struct stat sbuf;
 
-	fstat(fp->_file, &sbuf);
+	fstat(fp, &sbuf);
```

Comment:
This line is the culprit. glibc2 fstat() takes a FILE pointer directly, and FILE objects are opaque under glibc2, so they can't be inspected in the way fstat() here is trying to do.